### DISCUSSION TITLE
"Allow specified actors to bypass required pull requests" only applies to repo admins

### DISCUSSION THREAD

**Discussion by dhermes** [community/community#51495] [Repositories]
### Select Topic Area

Bug

### Body

From **looking at** the feature (and the [docs][1]), it appears the goal is for it to work for non-admin users. So maybe it's actually a bug in GitHub. Some potential ways to interpret this as a bug:

- (A) Non-admin users **should** be able to "bypass required pull requests" (but they can't)
- (B) Non-admin users **should not** be able to be added to the "Allow specified actors to bypass required pull requests" list (but they can be)
- (C) There should be a warning if an (invalid) non-admin users is added to / present in the "Allow specified actors to bypass required pull requests" list (but there isn't)

See https://github.com/github/docs/pull/24726

[1]: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule#creating-a-branch-protection-rule

**Comment by elee-msft** [comment]
I'm blocked by this issue right now, because I need the Azure Pipelines app to be able to push directly to a protected branch, and the "Allow specified actors to bypass required pull requests" checkbox appears designed to allow exactly that scenario, but it simply does not work.

**Comment by patrick-knight** [comment]
Is the `Require linear history` or any other rule also enabled with the pull request bypass?

**Comment by steve-taylor** [comment]
This seemed to be working until recently. The bizarre behaviour I'm witnessing right now:

### Doesn't work

* ✅  Allow specified actors to bypass required pull requests
  * `some-github-team`, which contains the member `some-github-user` (fictional names for this discussion) is added to the list
* ✅  Require status checks to pass before merging
  * ✅  Require branches to be up to date before merging
  * `build` - name of status check, which is a GitHub Actions job

Error on `git push` from my release workflow:
```
! [remote rejected]     main -> main (protected branch hook declined)
```

Note: `git` is authenticated using a token generated by `some-github-user`, _not_ `GITHUB_TOKEN`.

### Works

All the above, except I remove `build` from the list of status checks, leaving the list of status checks empty.

The problem with this workaround is that it allows pull requests to be merged without the build passing.
